vtx_common/tools/verify_version.py

```diff
@@ -0,0 +1,25 @@
+import os
+import sys
+import tomllib
+
+# Ensure our CI tag value matches our pyproject.toml value in our current working directory.
+
+def main(argv):
+    envar = 'CIRCLE_TAG'
+    if argv:
+        envar = argv[0]
+    etag = os.getenv('CIRCLE_TAG', '')
+    etag = etag.lstrip('v')
+
+    with open('pyproject.toml', 'r') as fd:
+        data = tomllib.loads(fd.read())
+        ptag = data.get('project', {}).get('version', 'project.version missing')
+
+    if etag != ptag:
+        info = f"Git tag from {envar} => {etag} does not match the version of this app: {ptag}"
+        print(info)
+        return 1
+    return 0
+
+if __name__ == '__main__':
+    sys.exit(main(sys.argv[1:]))
```

Comment:
nit: pass the full `sys.argv` instead of the slice. It's confusing in `main()` when I see `argv[0]` and that's not the program name.

Reply:
We do quite a bit of arg slicing in mainline code. The only reason it was in here was to make some manual testing a little bit easier; and if we swap CI providers we're not hard coded into `CIRCLE_TAG` at this module.

Reply:
our convention is that we always do the slice :)